server/src/main/java/com/linecorp/centraldogma/server/CentralDogma.java

```diff
@@ -782,9 +782,24 @@ private static Function<? super HttpService, EncodingService> contentEncodingDec
                 .build(delegate);
     }
 
-    private void configureMetrics(ServerBuilder sb, PrometheusMeterRegistry registry) {
+    private void configureMetrics(ServerBuilder sb, MeterRegistry registry) {
         sb.meterRegistry(registry);
-        sb.service(METRICS_PATH, new PrometheusExpositionService(registry.getPrometheusRegistry()));
+
+        // expose the prometheus endpoint if the registry is either a PrometheusMeterRegistry or
+        // CompositeMeterRegistry
+        if (registry instanceof PrometheusMeterRegistry) {
+            final PrometheusMeterRegistry prometheusMeterRegistry = (PrometheusMeterRegistry) registry;
+            sb.service(METRICS_PATH,
+                       PrometheusExpositionService.of(prometheusMeterRegistry.getPrometheusRegistry()));
+        } else if (registry instanceof CompositeMeterRegistry) {
```

Comment:
Does it make sense to use `PrometheusMeterRegistry` if the `CompositeMeterRegistry` contains it?
For example:
https://github.com/line/armeria/blob/7626562204bd4311a1205df9d733203324355a96/spring/boot2-autoconfigure/src/main/java/com/linecorp/armeria/spring/PrometheusSupport.java#L40-L55

Reply:
I think it's fine to leave as is since there is almost no difference in behavior.

For instance, assuming a `PrometheusMeterRegistry` already exists (`PrometheusMeterRegistry1`).
When the `centraldogma` server is started up, `CompositeMeterRegistry` will have (`PrometheusMeterRegistry1`, `PrometheusMeterRegistry2`).
At this point, whatever metrics have been added to `CompositeMeterRegistry` will also be synchronized to both children. (although it is possible each metric has different states for a little while, it will eventually become consistent).

Since the meters for both registries will eventually become the same, I thought it wasn't a big deal (although there may be slight performance degradation)

I actually think the current `PrometheusSupport` implementation is slightly flawed in that children `CompositeMeterReigstry`s aren't accounted for. (I guess that using [CompositeMeterRegistry#nonCompositeDescendants ](https://github.com/micrometer-metrics/micrometer/blob/38003f544f8a1ceb29d9bc634e964d2606f8bb39/micrometer-core/src/main/java/io/micrometer/core/instrument/composite/CompositeMeterRegistry.java#L49) would be cleanest, but unforunately it seems like this isn't available publicly)

Reply:
Understood. Let's keep it as is.